src/pkg/packager/actions/actions.go

```diff
@@ -89,82 +90,55 @@ func runAction(ctx context.Context, defaultCfg v1alpha1.ZarfComponentActionDefau
+			return err
+		}
+		out = strings.TrimSpace(out)
 
-	// Keep trying until the max retries is reached.
-retryCmd:
-	for remaining := actionDefaults.MaxRetries + 1; remaining > 0; remaining-- {
-		// Perform the action run.
-		tryCmd := func(ctx context.Context) error {
-			// Try running the command and continue the retry loop if it fails.
-			if out, err = actionRun(ctx, actionDefaults, cmd, actionDefaults.Shell, spinner); err != nil {
+		// If an output variable is defined, set it.
+		for _, v := range action.SetVariables {
+			variableConfig.SetVariable(v.Name, out, v.Sensitive, v.AutoIndent, v.Type)
+			if err := variableConfig.CheckVariablePattern(v.Name, v.Pattern); err != nil {
 				return err
 			}
-
-			out = strings.TrimSpace(out)
-
-			// If an output variable is defined, set it.
-			for _, v := range action.SetVariables {
-				variableConfig.SetVariable(v.Name, out, v.Sensitive, v.AutoIndent, v.Type)
-				if err := variableConfig.CheckVariablePattern(v.Name, v.Pattern); err != nil {
-					return err
-				}
-			}
-
-			// If the action has a wait, change the spinner message to reflect that on success.
-			if action.Wait != nil {
-				spinner.Successf("Wait for \"%s\" succeeded", cmdEscaped)
-			} else {
-				spinner.Successf("Completed \"%s\"", cmdEscaped)
-			}
-
-			// If the command ran successfully, continue to the next action.
-			return nil
 		}
 
-		// If no timeout is set, run the command and return or continue retrying.
-		if actionDefaults.MaxTotalSeconds < 1 {
-			spinner.Updatef("Waiting for \"%s\" (no timeout)", cmdEscaped)
-			//TODO (schristoff): Make it so tryCmd can take a normal ctx
-			if err := tryCmd(context.Background()); err != nil {
-				continue retryCmd
-			}
-
-			return nil
+		// If the action has a wait, change the spinner message to reflect that on success.
+		if action.Wait != nil {
+			spinner.Successf("Wait for \"%s\" succeeded", cmdEscaped)
+		} else {
+			spinner.Successf("Completed \"%s\"", cmdEscaped)
 		}
 
-		// Run the command on repeat until success or timeout.
-		spinner.Updatef("Waiting for \"%s\" (timeout: %ds)", cmdEscaped, actionDefaults.MaxTotalSeconds)
-		select {
-		// On timeout break the loop to abort.
-		case <-timeout:
-			break retryCmd
-
-		// Otherwise, try running the command.
-		default:
-			ctx, cancel := context.WithTimeout(ctx, duration)
-			defer cancel()
-			if err := tryCmd(ctx); err != nil {
-				continue retryCmd
+		return nil
+	},
+		retry.Context(actionCtx),
+		retry.Attempts(uint(actionDefaults.MaxRetries)),
+		retry.DelayType(retry.FixedDelay),
+		retry.Delay(100*time.Millisecond),
+		retry.OnRetry(func(uint, error) {
+			timeoutStr := ""
+			if actionDefaults.MaxTotalSeconds > 0 {
+				timeoutStr = fmt.Sprintf("(timeout: %ds)", actionDefaults.MaxTotalSeconds)
 			}
-
-			return nil
-		}
+			spinner.Updatef("Waiting for \"%s\" %s", cmdEscaped, timeoutStr)
+		}),
+	)
+	if actionCtx.Err() != nil {
```

Comment:
This will also happen on context cancel so need another way of checking for timeout. An option is to use with cause.